components/forms/Form/Form.tsx

```diff
@@ -45,6 +49,36 @@ const InnerForm = (props: InnerFormProps & FormikProps<FormValues> & DynamicForm
     }
   }, [formStatusError, errorList, lastSubmitCount, canFocusOnError]);
 
+  useEffect(() => {
+    if (!formConfig) {
+      return;
+    }
+
+    // calculate initial delay for submit timer
+    const secondsBaseDelay = 2;
+    const secondsPerFormElement = 2;
+    const numberOfRequiredElements = formConfig.elements.filter(
+      (element) => element.properties.validation?.required === true
+    ).length;
+
+    const submitDelaySeconds = secondsBaseDelay + numberOfRequiredElements * secondsPerFormElement;
+    setSubmitDelay(submitDelaySeconds);
+    setSubmitTimer(submitDelaySeconds);
+  }, [formConfig]);
+
+  useEffect(() => {
+    // timeout to prevent form from being submitted too quickly
+    let timeoutId = 0;
+    if (submitTimer > 0) {
+      timeoutId = window.setTimeout(() => {
```

Comment:
I modified to `window.setTimeout` as it returns a number.  Instead of using `setTimeout()` which returns a type of NodeJS.Timer.  This will never be called server side (aka Node) so we're safe to use `window.setTimeout()` here.  It also saves us from retyping from NodeJS.Timer => unknown => number.